packages/shared/src/components/feeds/FeedSettings/sections/FeedSettingsTagsSection.tsx

```diff
@@ -0,0 +1,146 @@
+import { Origin } from '../../../../lib/log';
+import { TagElement } from '../TagElement';
+import {
+  ModalKind,
+  ModalPropsContext,
+  ModalSize,
+} from '../../../modals/common/types';
+import { ModalTabs } from '../../../modals/common/ModalTabs';
+import useDebounceFn from '../../../../hooks/useDebounceFn';
+import { useTagSearch } from '../../../../hooks/useTagSearch';
+import { SearchField } from '../../../fields/SearchField';
+import { TagsData } from '../../../../graphql/feedSettings';
+import {
+  Typography,
+  TypographyColor,
+  TypographyType,
+} from '../../../typography/Typography';
+import useFeedSettings from '../../../../hooks/useFeedSettings';
+import useTagAndSource from '../../../../hooks/useTagAndSource';
+
+const tabs = ['Suggested', 'My tags'];
+
+const noop = () => undefined;
+
+export const FeedSettingsTagsSection = (): ReactElement => {
+  const { feed, onTagClick } = useContext(FeedSettingsEditContext);
+  const [activeView, setActiveView] = useState<string>(() => tabs[0]);
+  const { feedSettings } = useFeedSettings({ feedId: feed?.id });
+  const { onUnfollowTags } = useTagAndSource({
+    origin: Origin.CustomFeed,
+    feedId: feed?.id,
+    shouldFilterLocally: true,
+    shouldUpdateAlerts: true,
+  });
+
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const [onSearch] = useDebounceFn(setSearchQuery, 200);
+
+  const { data: searchResult } = useTagSearch({
+    value: searchQuery,
+    origin: Origin.CustomFeed,
+  });
+
+  const onboardingTagsPerLetter = useMemo(() => {
+    const tagsToGroup = searchQuery
+      ? searchResult?.searchTags.tags
+      : feedSettings?.includeTags;
+
+    if (!tagsToGroup) {
+      return [];
+    }
+
+    const tagsPerLetter = tagsToGroup.reduce((acc, tagItem) => {
+      const tag = typeof tagItem === 'string' ? { name: tagItem } : tagItem;
+
+      const firstLetter = tag.name[0].toUpperCase();
+      acc[firstLetter] = acc[firstLetter] || [];
+      acc[firstLetter].push(tag);
+
+      return acc;
+    }, {} as Record<string, TagsData['tags']>);
+
+    const sortedTagsPerLetter = Object.entries(tagsPerLetter).sort(([a], [b]) =>
+      a.localeCompare(b),
+    );
+
+    return sortedTagsPerLetter;
+  }, [feedSettings, searchResult, searchQuery]);
+
+  return (
+    <div className="flex flex-col gap-6">
+      <SearchField
+        aria-label="Search tags"
+        className="border-none !bg-background-subtle"
+        inputId="search-filters"
+        placeholder="Search tags"
+        valueChanged={onSearch}
+      />
+      <ModalPropsContext.Provider
+        value={{
+          tabs,
+          activeView,
+          setActiveView,
+          onRequestClose: noop,
+          kind: ModalKind.FlexibleCenter,
+          size: ModalSize.Medium,
+        }}
+      >
+        <ModalTabs className="border-b border-border-subtlest-tertiary pb-[0.70rem]" />
+        <div className="flex w-full max-w-full flex-col">
+          {activeView === 'Suggested' && (
```

Comment:
Probably neat with Enum but not really urgent/needed

Reply:
yeah, will add